### ISSUE TITLE
The floating shadows have not been removed and the training results are very poor.

### ISSUE THREAD

**Issue by crazyzyz** [coderpiaobozhe/Low-Frequency-First-Eliminating-Floating-Artifacts-in-3D-Gaussian-Splatting#2]
Hi there, I used this command to train my dataset, but the results I got were like this. The effect was not good; it didn't effectively remove the floating shadows. Instead, there were many of them. I have removed the comment for "Low-Quality Initialization". According to my understanding, it should have followed your manual, but the result was very poor. Could you please give me some suggestions? 

python train.py -s /root/3DGS/data/your_scene_name -m /root/3DGS/output/my_scene --eval --white_background --init_scaling_multiplier_max 1.5 --init_scaling_multiplier_min 1.0 --interval_times 2 --diffscale True

<img width="438" height="687" alt="Image" src="https://github.com/user-attachments/assets/f043cd26-5641-4375-b432-dabfb97854d2" />

<img width="884" height="254" alt="Image" src="https://github.com/user-attachments/assets/c0b4250a-eb68-40f7-9b95-5c854d4c3b35" />

**Comment by jcwang-gh** [comment]
Thank you for trying our work! I’m currently in the process of cleaning up the code and datasets, so there may still be some bugs. That said, this is the first time I’ve encountered a situation like yours. Since you haven’t provided specific details about your dataset, I can’t pinpoint the exact issue, but I’d be happy to share some possible explanations:

1. It appears that your image might not be a rendered output but rather a screenshot from software like MeshLab. If that’s the case, the artifacts you’re observing could be far from the 3D object/scene you’re trying to reconstruct or might not be visible in most of the training camera views. If my assumption is correct, these artifacts wouldn’t participate in the optimization process, making them difficult to remove with our current method.
2. As mentioned in the README, tuning EFA-GS’s hyperparameters is crucial for achieving optimal performance across different datasets. You might want to experiment with different hyperparameter settings to see if that resolves the issue.

To help us better understand and address your case, could you kindly provide the following details?

1. What is the PSNR result in your case?
2. How does the original 3D Gaussian Splatting (3DGS) perform on your dataset?

Your feedback would be greatly appreciated, and I’d be happy to assist further once I have more information.

**Comment by crazyzyz** [comment]
> Thank you for trying our work! I’m currently in the process of cleaning up the code and datasets, so there may still be some bugs. That said, this is the first time I’ve encountered a situation like yours. Since you haven’t provided specific details about your dataset, I can’t pinpoint the exact issue, but I’d be happy to share some possible explanations:
> 
> 1. It appears that your image might not be a rendered output but rather a screenshot from software like MeshLab. If that’s the case, the artifacts you’re observing could be far from the 3D object/scene you’re trying to reconstruct or might not be visible in most of the training camera views. If my assumption is correct, these artifacts wouldn’t participate in the optimization process, making them difficult to remove with our current method.
> 2. As mentioned in the README, tuning EFA-GS’s hyperparameters is crucial for achieving optimal performance across different datasets. You might want to experiment with different hyperparameter settings to see if that resolves the issue.
> 
> To help us better understand and address your case, could you kindly provide the following details?
> 
> 1. What is the PSNR result in your case?
> 2. How does the original 3D Gaussian Splatting (3DGS) perform on your dataset?
> 
> Your feedback would be greatly appreciated, and I’d be happy to assist further once I have more information.

The tool I used is called "supersplat", which is also a very good tool for displaying gs ply. Therefore, there is no issue of not being a rendered output. Meshlab is unable to effectively display Gaussian data. I did not try any hyperparameters. Later, I can try the same thing. The original 3DGS will be better than ours in terms of effect. However, my main intention is to try to remove the floating shadows in our project.

**Comment by jcwang-gh** [comment]
Thank you for your patience. I just noticed that you’ve been using the "low-quality initialization" option. As demonstrated in our paper, this mode typically yields significantly poorer results compared to the standard normal initialization, which is why we’ve set normal initialization as the default.

The low-quality initialization option is primarily included to allow users to reproduce the comparative results discussed in the paper. For optimal performance with your dataset, I’d highly recommend switching to normal initialization (to obtain COLMAP results) and giving it another try. This adjustment may substantially improve your outcomes.

Please don’t hesitate to let me know if you encounter any further issues—I’d be happy to help!

**Comment by crazyzyz** [comment]
> Thank you for your patience. I just noticed that you’ve been using the "low-quality initialization" option. As demonstrated in our paper, this mode typically yields significantly poorer results compared to the standard normal initialization, which is why we’ve set normal initialization as the default.
> 
> The low-quality initialization option is primarily included to allow users to reproduce the comparative results discussed in the paper. For optimal performance with your dataset, I’d highly recommend switching to normal initialization (to obtain COLMAP results) and giving it another try. This adjustment may substantially improve your outcomes.
> 
> Please don’t hesitate to let me know if you encounter any further issues—I’d be happy to help!

OMG, sorry, I did it this way. Do you mean that a normal initialization can produce better results than the current ones and can effectively remove the floating shadows?

**Comment by crazyzyz** [comment]
> Thank you for your patience. I just noticed that you’ve been using the "low-quality initialization" option. As demonstrated in our paper, this mode typically yields significantly poorer results compared to the standard normal initialization, which is why we’ve set normal initialization as the default.
> 
> The low-quality initialization option is primarily included to allow users to reproduce the comparative results discussed in the paper. For optimal performance with your dataset, I’d highly recommend switching to normal initialization (to obtain COLMAP results) and giving it another try. This adjustment may substantially improve your outcomes.
> 
> Please don’t hesitate to let me know if you encounter any further issues—I’d be happy to help!

I will now conduct the test under normal circumstances. However, since my test set only contains the "sparse/0" entry, I discovered from the convert code that this might not have any impact on my results. This is because I only have a single 0 file and there are no multiple folders.

**Comment by jcwang-gh** [comment]
> > Thank you for your patience. I just noticed that you’ve been using the "low-quality initialization" option. As demonstrated in our paper, this mode typically yields significantly poorer results compared to the standard normal initialization, which is why we’ve set normal initialization as the default.
> > The low-quality initialization option is primarily included to allow users to reproduce the comparative results discussed in the paper. For optimal performance with your dataset, I’d highly recommend switching to normal initialization (to obtain COLMAP results) and giving it another try. This adjustment may substantially improve your outcomes.
> > Please don’t hesitate to let me know if you encounter any further issues—I’d be happy to help!
> 
> OMG, sorry, I did it this way. Do you mean that a normal initialization can produce better results than the current ones and can effectively remove the floating shadows?

Yes. In our previous experiments, the results of normal initialization are usually much better than low-quality initialization. That's why we name the setting 'low-quality initialization'😂. We use low-quality setting just to create more artifacts for experimental use. In the most case you should use normal initialization to achieve better results. I'm sorry that these settings were not clearly explained in README, which may have caused some confusion. I'll update the documentation immediately to clarify this point.

If you have any other questions about our work, feel free to ask and I'm glad to help.

**Comment by jcwang-gh** [comment]
> > Thank you for your patience. I just noticed that you’ve been using the "low-quality initialization" option. As demonstrated in our paper, this mode typically yields significantly poorer results compared to the standard normal initialization, which is why we’ve set normal initialization as the default.
> > The low-quality initialization option is primarily included to allow users to reproduce the comparative results discussed in the paper. For optimal performance with your dataset, I’d highly recommend switching to normal initialization (to obtain COLMAP results) and giving it another try. This adjustment may substantially improve your outcomes.
> > Please don’t hesitate to let me know if you encounter any further issues—I’d be happy to help!
> 
> I will now conduct the test under normal circumstances. However, since my test set only contains the "sparse/0" entry, I discovered from the convert code that this might not have any impact on my results. This is because I only have a single 0 file and there are no multiple folders.

emmm, that's weird. Maybe 'convert.py' has some bugs. If you have ever tried original 3DGS on this dataset, you can use the colmap result of the original 3DGS project because theoretically normal initialization is the same setting as the one used in original 3DGS project.
